workflows/escherichia_coli.nf

```diff
@@ -26,16 +26,20 @@ include { virulencefinder                           } from '../nextflow-modules/
 include { CALL_BACTERIAL_BASE                       } from '../workflows/bacterial_base.nf'
 
 workflow CALL_ESCHERICHIA_COLI {
+    // Create channel for sample metadata
     Channel.fromPath(params.csv)
         .splitCsv(header:true)
         .map{ row -> get_meta(row) }
-        .branch {
-            iontorrent: it[2] == "iontorrent"
-            illumina: it[2] == "illumina"
-            nanopore: it[2] == "nanopore"
-        }
+        .map{ row -> [row[0], row[2]] }
         .set{ ch_meta }
 
+    // Create channel for reads
```

Comment:
Is there not a way to do this without having to read the csv file twice?

Reply:
Yes there are but I think that requires setting temporary variables.